### PR TITLE
LTD-2035: Add ui test that verifies enforcement check by importing the file

### PR DESCRIPTION
**Pull request by saruniitr**
## Change description

Enforcement checks are done on parties of an application. We first export
the data in an xml file and after the checks are performed another file
is imported which contains data to update the application based on the
checks.

In this test we generate the file to be imported based on the parties
data and once imported we assert that the application is updated or not.

### CODE REVIEW

File: ui_tests/caseworker/step_defs/test_enforcement.py [hunk 1]
```diff
@@ -19,6 +19,11 @@ def export_enforcement_xml(driver):
     CaseListPage(driver).click_export_enforcement_xml()
 
 
+@when(parsers.parse('I press "{export_eu_btn_text}"'))
+def export_enforcement_xml(driver, export_eu_btn_text):
+    CaseListPage(driver).click_by_link_text(export_eu_btn_text)
+
```
Comment (wkeeling) [inline ui_tests/caseworker/step_defs/test_enforcement.py:25]: There's a general purpose step for pressing buttons - can that be used here?
```
When I click "button text"
```
(the underlying method is `click_button_with_text()`).

File: ui_tests/caseworker/features/enforcement.feature [hunk 2]
```diff
@@ -40,19 +40,19 @@ Feature: I want to export and import XML for enforcement checking
...
     And I go to application previously created
     Then the application is removed from "Enforcement Unit Cases to Review" queue
-    And the flag "Enforcement Check Req" is not present
     And I cleanup the temporary files created
 
     Examples:
-    | name    | product | part_number | clc_rating  | end_user_name      | end_user_address  | country | consignee_name      | consignee_address   | end_use                  |
-    | Test    | Rifle   | ABC-123-45  | PL9002      | Automated End user | 1234, High street | BE      | Automated Consignee | 1234, Trade centre  | Research and development |
+    | name    | product        | part_number | clc_rating  | end_user_name | end_user_address| country | consignee_name    | consignee_address | end_use                  |
+    | Test    | Rifle1, Rifle2 | PN-ABC-123  | PL9002      | Joe bloggs    | 123 Main street | BL      | Josephine Bloggs  | 123 Main Street   | Research and development |
```
Comment (wkeeling) [inline ui_tests/caseworker/features/enforcement.feature:58]: Missing a new line by the looks of it

File: ui_tests/caseworker/step_defs/test_enforcement.py (hunk 1 above)
Comment (saruniitr) [reply ui_tests/caseworker/step_defs/test_enforcement.py:25]: yes, just went by the bdd text

File: ui_tests/caseworker/step_defs/test_enforcement.py [hunk 3]
```diff
@@ -73,9 +83,45 @@ def generate_enforcement_check_import_file(party_type):
...
+    root = tree.getroot()
+
+    # compare the values in the file to be imported for the given party
+    for index, node in enumerate(root.findall(f".//SPIRE_RETURNS/{import_tag}")):
+        assert node.text == data[index][export_tag]
+
+
+@when(parsers.parse('I click on "{import_eu_btn_text}"'))
+def import_enforcement_xml(driver, import_eu_btn_text):
+    CaseListPage(driver).click_by_link_text(import_eu_btn_text)
```
Comment (wkeeling) [inline ui_tests/caseworker/step_defs/test_enforcement.py:120]: As above, maybe use the general button click step.

File: ui_tests/caseworker/features/enforcement.feature [hunk 4]
```diff
@@ -40,19 +40,19 @@ Feature: I want to export and import XML for enforcement checking
...
-    And the downloaded file should include "CONSIGNEE" "COUNTRY" as "Belgium"
-    And the downloaded file should include "END_USER" "COUNTRY" as "Belgium"
-    When I include "CONSIGNEE" details to generate import file
+    And I press "Export EU XML"
+    Then an XML file is downloaded onto my device
+    When I include "END_USER" details and generate import file
+    And for FLAG the file has "N"
+    And for "CODE1" the file has the "END_USER" data "ELA_ID" number from export file
+    And for "CODE2" the file has the "END_USER" data "SH_ID" number from export file
     And I go to my case list
```
Comment (wkeeling) [inline ui_tests/caseworker/features/enforcement.feature:49]: This should probably be a `When` step, since it's starting a new sequence of navigation.

**Review by wkeeling — APPROVED** [review]

**Pull request merged**